homeassistant/components/tfl/config_flow.py

```diff
@@ -0,0 +1,266 @@
+                self.hass, user_input[CONF_API_APP_KEY]
+            )
+
+            if not validation_result.errors:
+                # Input is valid, set data
+                self.data = user_input
+                self.options[CONF_STOP_POINTS] = []
+                return await self.async_step_stop_point()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=STEP_USER_DATA_SCHEMA,
+            errors=validation_result.errors,
+        )
+
+    async def async_step_stop_point(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the stop point step."""
+        validation_result = ValidationResult()
+        if user_input is not None:
+            app_key = self.data[CONF_API_APP_KEY]
+            validation_result = await validate_stop_points(
+                self.hass, app_key, user_input[CONF_STOP_POINTS]
+            )
+
+            if not validation_result.errors:
+                # User input is valid, save the stop point
+                self.options[CONF_STOP_POINTS].extend(user_input[CONF_STOP_POINTS])
+
+                # Create the entry
+                return self.async_create_entry(
+                    title="Transport for London", data=self.data, options=self.options
+                )
+
+        return self.async_show_form(
+            step_id="stop_point",
+            data_schema=STEP_STOP_POINTS_DATA_SCHEMA,
+            errors=validation_result.errors,
+            description_placeholders=validation_result.description_placeholders,
+        )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Reconfigure step allows the app key to be updated."""
+        validation_result = ValidationResult()
+        config_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        assert config_entry is not None
+        if user_input is not None:
+            app_key = user_input.get(CONF_API_APP_KEY, "")
+            validation_result = await validate_app_key(self.hass, app_key)
+
+            if not validation_result.errors:
+                # Input is valid, set data
+                self.data[CONF_API_APP_KEY] = app_key
+                self.hass.config_entries.async_update_entry(
+                    config_entry, data=self.data
+                )
+                return self.async_abort(reason="reconfigure_successful")
+
+        return self.async_show_form(
+            step_id="reconfigure",
+            data_schema=self.add_suggested_values_to_schema(
+                RECONFIGURE_DATA_SCHEMA,
+                config_entry.data if user_input is None else user_input,
+            ),
+            errors=validation_result.errors,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
```

Comment:
Let's leave options and reconfigure flow out for now for the scope and size of the PR

Reply:
Sure thing. Keen to get this PR over the line, so happy to do this; makes sense.

Reply:
Done in e591fe286e29db2843826580bd62c19279722202